src/server/mod.rs

```diff
@@ -993,13 +1004,53 @@ impl LanguageServer for LspServer {
                     _ => None,
                 });
             if let Some(typ) = hover_type {
+                let supports_markdown = match self
+                    .client_capabilities
+                    .read()
+                {
+                    Ok(client_capabilities) => {
+                        if let Some(text_document) =
+                            (*client_capabilities)
+                                .text_document
+                                .as_ref()
+                        {
+                            if let Some(hover) =
+                                text_document.hover.as_ref()
+                            {
+                                hover.content_format.as_ref().map_or(false, |formats| formats.contains(&lsp::MarkupKind::Markdown))
+                            } else {
+                                false
+                            }
+                        } else {
+                            false
+                        }
+                    }
+                    Err(err) => {
+                        log::error!("{}", err);
+                        false
+                    }
+                };
+                let hover_contents: lsp::HoverContents =
+                    match supports_markdown {
+                        true => lsp::HoverContents::Markup(
+                            lsp::MarkupContent {
+                                kind: lsp::MarkupKind::Markdown,
+                                value: format!(
+                                    "```flux\ntype: {}\n```",
```

Comment:
```suggestion
                                    "```flux\n{}\n```",
```
The markdown expects to be valid flux so I don't think the `type:` prefix should remain (it could cause weird syntax highlighting for instance, since the syntax isn't valid).

If we do want to describe it as the type we could move that outside the codeblock instead (where we might also add other things, like the doc comment).

(Example)

    ### Docs
    etc
    
    ### Type
    ```flux
    {}
    ```

Reply:
Got it! I saw you didn't put `type` string in when you changed the code from `Scalar` to `Markdown` and I thought it was a typo 😅 Now it make sense to me why you intentionally didn't put it in